### DISCUSSION TITLE
Error: cannot_register

### DISCUSSION THREAD

**Discussion by wahur666** [processone/ejabberd#3584] [Questions] [answered]
Hi,

I'm trying to setup ejabberd on a clean VM Ubuntu server 20.04, and this "very descriptive" error message cames up. Error: cannot_register. No other log inside the live running instance.

Steps to reproduce:
1. Install it from the .deb file
2. Try to register user

Thanks for the help.
![kép](https://user-images.githubusercontent.com/9078042/115685838-e33ff080-a358-11eb-8794-b52a89d3b2d6.png)
![kép](https://user-images.githubusercontent.com/9078042/115685895-f2bf3980-a358-11eb-8f6c-c4cf98338dbf.png)

**Comment by prefiks** [comment, marked as answer]
You need to use host that is defined in your config file (i don't think localhost or example.org is one of them, i guess it's imre-virtualbox? but you need to see what you have in hosts section in ejabberd.yml).

**Reply by wahur666** [reply]
Thank you, for the quick response.

I'm following this page: https://docs.ejabberd.im/admin/installation/#install-with-os-specific-packages

Yes, that was the issue, that I don't used the proper host.  But I want to ask, why it is not written somewhere in the logs? cannot register is not saying anything what was the issue. 

On the other hand maybe this also documented, but as you can see, my host contains capitals,  and the jabberd id uses only lower cases, this also was a point where the admin page was not accessible. 

Maybe I ran into to many corner cases, but would be useful a mention here and there, at least for first time users, why it is not working.

But thanks after all :)

**Reply by prefiks** [reply]
Error that is generated has reason field in it it explains this, but for some reason ejabberdctl don't display it, will need to see why it's not shown.

**Reply by prefiks** [reply]
I commited 5a8f1ca528e0272fb4911062d718ef7a975d8325 which show better reason in this case.
